nn_modules/decoder/decoder.py

```diff
@@ -0,0 +1,40 @@
+'''
+Author: diwei diwei@qcraft.ai
+Date: 2024-06-15 12:32:58
+LastEditors: diwei diwei@qcraft.ai
+LastEditTime: 2024-06-15 17:21:11
+FilePath: /top_eleven/nn_modules/decoder/decoder.py
+Description: 
+
+Copyright (c) 2024 by ${git_name_email}, All Rights Reserved. 
+'''
+from torch import nn
+from torch import Tensor
+from typing import Optional
+
+
+class TopDecoder(nn.Module):
+    def __init__(self, d_model: int, nhead: int) -> None:
+        super().__init__()
+        self.decoder_layer = nn.TransformerDecoderLayer(
+            d_model, nhead, batch_first=True)
+
+    def forward(self, x: Tensor, memory: Tensor, mask: Optional[Tensor] = None) -> Tensor:
```

Comment:
Not sure if we really need the mask in our specific problem.
My opinion is that if it is not used, then we could simply omit it.